packages/components/src/Chip/components/ChipSelectable/ChipSelectable.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from "react";
+import { Chip } from "../../Chip";
+import { ChipSelectableProps } from "../../Chip.types";
+import styles from "../../Chip.css";
+import { Icon } from "../../../Icon";
+
+export function ChipSelectable({
+  ariaLabel,
+  disabled,
+  heading,
+  invalid,
+  label,
+  selected,
+  role,
+  tabIndex,
+  value,
+  variation,
+  onClick,
+  onKeyDown,
+}: ChipSelectableProps) {
+  const [isChipSelected, setIsChipSelected] = useState(selected);
+
+  useEffect(() => {
+    setIsChipSelected(selected);
+  }, [selected]);
```

Comment:
now that the nitpicks are out of the way, should this be handled internally?

I think this is better handled through the consumer side of things. Keep the components reactive to props rather than it having a state and a side effect for the prop change.

Right now it just works which gives you a false sense of it actually saving the "selected" state only to find that it actually wasn't since the chip is doing something on its own rather than being controlled by the consumer's state.

Reply:
These are mainly convenience Chips to cut down on having to repeat patterns, so in this case I wanted to make the component as low stakes as possible to start using. So it will operate on its own without a consumer providing state, but it will also react to state changes in the event that you pass in a prop programatically (due to something else happening in the UI).

So in my head it's the best of both worlds. You can just use the component without having to worry about state management, but if you wanted to provide your own state for any reason, the component handles that as well.

Reply:
> You can just use the component without having to worry about state

Reply:
>  You can just use the component without having to worry about state

In theory, that makes sense. Until you get to a point where you would have to save that state of the chip somewhere so when you come back to it, it'll still have to correct state since it will always start off as false until you set that prop to true. Which then keeps on toggling since you have a state that updates it now.

In the end, there's always a state somewhere being saved that controls this Chip's appearance. So Chip having its own state for selected is redundant and possibly invites misconfiguration that leads to bugs.

It's the same thing that react does for the raw HTML where the moment you put in a `value` prop but don't properly implement the `onChange` to save that input's value in the state, it'll stop working until you've actually implemented it correctly. In react's POV, it's much easier to spot if that component just _reacts_ to what you've specifically told it to do than it doing something on its own until you realize you've made a mistake of not saving its state.

Reply:
I disagree with your premises outlined above but in the interest in getting this PR merged, I've made this requested change and we can have a follow-up discussion at some point in the future about this convenience pattern.

Reply:
sounds good. happy to talk about this in detail since there are legit use cases of a component having its own state. Like opening/closing a menu where the consumer doesn't need to worry about controlling that interaction. Compared to a state that needs to be synced between a parent and a child where an internal state might not be needed. It really depends on the use case of that state.